### PR TITLE
Migrate Degree Wizard to React Query

### PR DESCRIPTION
**Pull request by lhvy**
There are currently errors in the degree wizard merged into CIRCLES-331 that makes testing other branches somewhat difficult. This PR is a ~~quick~~ attempt to fix the ones stopping the front page from even loading.

The React Query migration for the Degree Wizard is almost entirely covered by this PR. This should also cover all the changes in #1060 at the same time.

### CODE REVIEW

File: frontend/src/pages/DegreeWizard/YearStep/YearStep.tsx [hunk 1]
```diff
@@ -16,14 +14,29 @@ const RangePicker = React.lazy(() =>
...
+  const handleOnChange = async (
+    _: RangeValue<dayjs.Dayjs>,
+    [startYear, endYear]: [string, string]
+  ) => {
+    // We can trust num years to be a valid number because the range picker only allows valid ranges
+    setDegreeInfo((prev) => ({
+      ...prev,
+      start_year: parseInt(startYear, 10),
+      end_year: parseInt(endYear, 10)
+    }));
```
Comment (Peedee2002) [inline frontend/src/pages/DegreeWizard/YearStep/YearStep.tsx]: Bingo

Comment (lhvy) [reply frontend/src/pages/DegreeWizard/YearStep/YearStep.tsx]: Thanks!

**Comment by lhvy** [comment]
Almost there! Just a few small fixes with input validation in the new API route to handle now, and a good review from @leonardo-fan and @imagine-hussain.

**Comment by imagine-hussain** [comment]
More changes made:
- Added actual validation to BE
- entire wizard is 1 transaction
- FE waits on BE to be correct before moving on

# Future implications (Out of Scope)

- Wizard should be locked behind login
- Add proper redirects (from other pages) to wizard if degree is not complete

**Review by imagine-hussain — APPROVED** [review]

**Comment by imagine-hussain** [comment]
Tests will fail due to changes - tests are out of date - aactual thing builds.
Is blocking others so, will merge in the changes and deal w/ tests later.

**Pull request merged**